### DISCUSSION TITLE
Case insensitive search with multiple fields

### DISCUSSION THREAD

**Discussion by ravargash** [hasura/graphql-engine#9069] [Question] [answered]
Hello again!

I'm trying to make a "smart search" where, using a single term in a single query, replies with all the similar results. Let's suppose we have this schema: 

![193170077-0fa6048d-fab7-4a6f-88b7-edeee40fad9c (1)](https://user-images.githubusercontent.com/27013133/193909264-54641f65-0e43-4ffe-8e7d-fbdfb024b3e7.png)

And I want to get results based on its name, brand or services. The query would be something like this:

<img width="292" alt="Screen Shot 2022-10-04 at 16 28 04" src="https://user-images.githubusercontent.com/27013133/193908543-6081b6f5-ecff-4e07-a701-d1ea13ec8e6e.png">

It does work! Cool and all, however, this search is not case insensitive, so its application is a bit janky.

How should I approach this?

Regards!

**Comment by adas98012** [comment, marked as answer]
You can use "_ilike" operator instead of "_similar":

[Reference](https://github.com/hasura/graphql-engine/issues/1516)

Here is a sample query and response:

```
query MyQuery2 {
  Album(where: {Title: {_ilike: "%For%"}}) {
    ArtistId
    Title
  }
}
```

```
"data": {
    "Album": [
      {
        "ArtistId": 1,
        "Title": "For Those About To Rock We Salute You"
      },
      {
        "ArtistId": 82,
        "Title": "King For A Day Fool For A Lifetime"
      },
      {
        "ArtistId": 88,
        "Title": "Appetite for Destruction"
      },
```
